stdlib/tkinter/__init__.pyi

```diff
@@ -392,11 +392,21 @@ class Misc:
     # The cnf parameter's default value is an empty dict. That's a perfectly
     # valid instance of _GridIndexInfo, but MyPy declares that an empty dict
     # literal must be dict[Any, Any] and is therefore incompatible.
+    #
+    # The **kw parameter must be a _GridIndexInfo too, but the type sanity-
+    # checker only permits us to declare one type for all kwargs values, instead
```

Comment:
What does "the type sanity-checker" mean? All of the tools we run in CI (e.g., stubtest, mypy, pyright) have friendly authors who would probably be receptive to making changes.

If I understand this correctly, though, we should just declare keyword arguments for every member of `_GridIndexInfo`, and `# type: ignore` any tool that insists we put `**kwargs`.

Reply:
There are a lot of tools run in CI, and I forget exactly which one complained about this particular pattern. I think it might have been stubtest?

At any rate, I'm not convinced the tool is wrong. In any sensible Python code, every value of a `**kwargs` dict *should* be the same type. It's just that tkinter is not sensible Python code, it's a very thin marshalling layer between Python and Tcl. For example, many tkinter method implementations are morally equivalent to:

```python
class SomeClass:
    def somemethod(self, **kwargs):
        args = [f"{k}  {v}" for k, v in kwargs.values()]
        self.tk.call(self.tk_name, "somemethod", *args)
```

...and any sanity check that requires the type signature's arguments to match the runtime code's arguments is going to stumble.

> ...and `# type: ignore` any tool that insists we put `**kwargs`.

I put the keyword arguments back in the `.pyi` file and added `# type: ignore` to each one, but that didn't quiet stubtest. I added the names to `tests/stubtest_allowlists/py3_common.txt` and that hushed it up, though.